go-controller/pkg/ovn/ovn.go

```diff
@@ -863,6 +864,20 @@ func (oc *Controller) WatchNodes() {
 				}
 				gatewaysFailed.Store(node.Name, true)
 			}
+
+			// ensure pods that already exist on this node have their logical ports created
+			options := metav1.ListOptions{FieldSelector: fields.OneTermEqualSelector("spec.nodeName", node.Name).String()}
+			pods, err := oc.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), options)
+			if err != nil {
+				klog.Errorf("Unable to list existing pods on node: %s, existing pods on this node may not function")
+			} else {
+				for _, pod := range pods.Items {
+					if !oc.ensurePod(nil, &pod, true) {
```

Comment:
`ensurePod` returns false for cases where the Pod is a HostNetwork Pod, so, we shouldn't be really retrying.

Furthermore, there is no guarantee that adding back the just deleted node will result in the node having the same HostSubnet CIDR. So, the Pods will have incorrect IPs.